hack/check-version.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+# Get the current git branch
+branch=$(git rev-parse --abbrev-ref HEAD)
+
+# Check if the branch is "main"
+if [[ "$branch" == "main" ]]; then
+    version="latest"
+# Otherwise, use the branch name as the version
+else
+    version="$branch"
+fi
+
+# Use yq to set build.version to version variable
+yq eval ".build.version = \"$version\"" -i build.yaml
```

Comment:
instead of the name of the file `check-version.sh` , I would call this `write-version.sh` or `commit-version.sh` or something consistent with what it is doing.

Reply:
why do we need a (git checked) file that contains the branch name (calling `latest` to `main`)?

Reply:
Hey, @eguzki ! I will leave your last comment for @guicassolato who could probably answer it best. 

As for your first comment, makes sense. Will do !

Reply:
> why do we need a (git checked) file that contains the branch name 

The idea of this file is to store all info needed to build the component that typically is passed as argument to the build command tools, thus avoiding otherwise hard-coding this info within the source, nor losing it somewhere within the Makefile. IOW, it's to separate build _data_ from build _logic_.

By "build" I mean the binary as well as the container image.

As proposed [here](https://github.com/Kuadrant/authorino/pull/473#issuecomment-2273351222), the `version` info assumes one of 3 possible values:
1. `latest` → for the latest builds based on the `main` branch – it should be constantly moving with the new SHAs being pushed to main
2. `<branch-name>` → for short-lived builds based on feature branches, mainly triggered manually during development, for testing, etc – typically lasting until a PR merged is only
3. `X.Y.Z` → for long-lived, stable builds based on release tags (named `vX.Y.Z`) – typically associated with one SHA that will never change

I guess 2 is the tricky one here. We want to be able to build binaries and tag container images after a branch name, so we identify these artifacts properly during dev/testing. However, this particular `version` info will only last until the feature branch gets merged. So, we do not want to commit `version: <branch-name>`. We want to brand the (short-lived) artifact like that, but without risking merging it. Instead, we want to commit `version: latest` (or whatever base the feature branch is targeting.)

Make sense?

Reply:
I guess in order to achieve that, we could set the version for building, and after build, revert the change. 

The downside with that is the user would not be able to see what their build info was post-build. 

Another solution might be to have a `recent` build info file which has all the info used to build at that time, and could be added to `.gitignore` since it would only exist post-build i.e. it would not be there before build. 

This would allow the user to see their most recent build config, and would allow for developers not to worry about git tracking unnecessary changes.

Reply:
This `version` information needs to be checked in git. Some projects do in the makefile, others do in `version.[go|java|..] and yet another approach is the build.yaml.  I do not really care about "how".  The "what" is what matters: version is specified somewhere. Consistency has a value, but can be sacrificed for greater good. 

All projects need some (documented) manual procedure or automated script to update the version and commit it.

What it does not make sense is to change is when image is building or executing "run" command. Image building or execute run command read build.yaml and with ldflags the information is embedded in the binary, available for future reference.

Reply:
> What it does not make sense is to change is when image is building or executing "run" command. Image building or execute run command read build.yaml and with ldflags the information is embedded in the binary, available for future reference.

Completely agree. The `build.yaml` should be the input here, not the output.

In general, the value of `version` inside `build.yaml` should be statically set to `latest`. I can see value in allowing make targets to use `build.yaml` or override it "on-the-fly" with something like the name of the branch, when running the workflow from a PR (via GHA) or locally, but without touching `build.yaml` then.

Separately, we need to update the [release process](https://github.com/Kuadrant/authorino/blob/main/RELEASE.md) to now include:
1. creating a `release-vX.Y.Z` branch
2. updating and commit `build.yaml` in the branch
3. Tag the release branch instead of `main`

We've never needed `release-vX.Y.Z` branches in Authorino and have always managed to simply tag a commit in `main` to cut a release, but that was because the version number never had to be committed along with the code before.

Reply:
OK, so I will go ahead with ensuring the build.yaml `.version` is statically kept as `latest` , having a recent build file ignored in `.gitignore`, and update the release process with this new information.